core/parentchain/indirect-calls-executor/src/executor.rs

```diff
@@ -122,7 +125,27 @@ impl<
 		debug!("Scanning block {:?} for relevant xt", block_number);
 		let mut executed_calls = Vec::<H256>::new();
 
-		for xt_opaque in block.extrinsics().iter() {
+		let events = self
+			.node_meta_data_provider
+			.get_from_metadata(|metadata| {
+				let raw_metadata: Metadata = metadata.clone().into()?;
+				Some(Events::<H256>::new(raw_metadata, block_hash, events.to_vec()))
+			})?
+			.ok_or_else(|| Error::Other("Could not unpack Events from Metadata".into()))?;
+
+		let xt_statuses = EventFilter::get_extrinsic_statuses(events)?;
+		debug!("xt_statuses:: {:?}", xt_statuses);
+
+		// This would be catastrophic but should never happen
+		if xt_statuses.len() != block.extrinsics().len() {
+			return Err(Error::Other("Extrinsic Status and Extrinsic count not equal".into()))
+		}
+
+		for (xt_opaque, xt_status) in block.extrinsics().iter().zip(xt_statuses.iter()) {
+			if let ExtrinsicStatus::Failed = xt_status {
+				continue
+			}
```

Comment:
Can we be certain that the order of the xts and the xt stati are the same?

Reply:
I still vote for doing this, as this is a small effort, which I prefer not to postpone. I want that down-stream implementors have less friction when rebasing.